packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -52,6 +52,41 @@ abstract class EndpointDispatch {
     return connector;
   }
 
+  Future<(MethodConnector, Map<String, dynamic>)> tryGetEndpointMethod({
+    required Session session,
+    required String endpointPath,
+    required String methodName,
+    required String body,
```

Comment:
Question: `body` here is really the serialized parameters, right? This layer doesn't really need to know the "body" concept since that more feels like a keyword in the HTTP transport layer.

Reply:
Very true, lets find a better name.

Reply:
Payload or data? Most important to call it the same everywhere!

Reply:
> It is not really reflected in the name of the method we are calling

I did update the name to clarify that an authorization takes place, but at some point I think a line has to be drawn. It seems unreasonable to cover every single thing that is happening in this method in its name. The goal I feel should be to cover the clear intent of the method and not do anything unexpected. As part of getting the method, a validation is made that it's a legit call. This seems fully expected to me, but this is somewhat subjective naturally! If you still think so, please give me a code example so I can see the full idea you had.

>  parameters should really be a concern when fetching the endpoint method?

This logic that we may want to update and should be true for all code paths. To me it seems most logical (and expected to future developers) to do it at the place of retrieval, but we can of course move it if we think that would make it even easier to understand and follow these code paths. But again, the goal here is to have a single entry point into all of this business logic.

Reply:
[resolved offline]: Agree with @SandPod to return a call context to make it clearer ([8573727](https://github.com/serverpod/serverpod/pull/2676/commits/8573727c0e0f74a5ceda85e4d518ae9a005b737b)), and also rename this and remove "Authorize" to make it clearer that a lot of things is happening and not only a single thing.